examples/logging.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use druid::shell::{runloop, WindowBuilder};
+use druid::widget::Column;
+use druid::{UiMain, UiState};
+
+fn main() {
+    init().ok(); // We init the simple logger here
+    druid::shell::init();
+
+    let mut run_loop = runloop::RunLoop::new();
+    let mut builder = WindowBuilder::new();
+    let col = Column::new();
+    let state = UiState::new(col, 0u32);
+    builder.set_title("Logging example");
+    builder.set_handler(Box::new(UiMain::new(state)));
+    let window = builder.build().unwrap();
+    window.show();
+    run_loop.run();
+}
+
+// Typically the user would use a crate for logging
+// Here we implement the Log trait for a struct and print everything to STDOUT
+// See https://docs.rs/log/ for more info
+use log::{Level, Metadata, Record};
```

Comment:
I appreciate the thoroughness, but I wonder if this isn't out of scope for this PR?

What I would personally be inclined to do is to find the simplest, most minimal logging crate ([simple_logger]](https://github.com/borntyping/rust-simple_logger) was the best i could find in a quick search) and just use that? It's closer to what the user will ever actually do.

Something we might consider for follow-up work is to add something like this to druid itself, and then we could let the user ask for stderr logging when they start the runloop.

Reply:
I wrote it like that to avoid any extra dependencies. Little did I know about [this](https://doc.rust-lang.org/rust-by-example/testing/dev_dependencies.html)...

So I added it as a dev-dependency.
Anyway, sounds good. I'll leave more concrete logging implications on the druid library up to another PR.

Reply:
yea, dev dependency is perfect. 👍